### ISSUE TITLE
Unable to resolve path to module 'katex' with eslint-import-resolver-webpack

### ISSUE THREAD

**Issue by andersk** [import-js/eslint-plugin-import#2293]
`import katex from "katex";` is valid according to both native Node.js and webpack. But eslint-import-resolver-webpack is unable to resolve it, leading to a spurious error `Unable to resolve path to module 'katex'`. This might have something to do with the `exports` map in [`katex/package.json`](https://unpkg.com/browse/katex@0.15.1/package.json).

Complete reproducible example:

**`src.js`**

```js
import katex from "katex";
console.log("Hello KaTeX", katex.version);
```

**`.eslintrc.json`**

```json
{
  "env": {
    "es6": true
  },
  "parserOptions": {
    "sourceType": "module"
  },
  "plugins": ["import"],
  "rules": {
    "import/no-unresolved": "error"
  },
  "settings": {
    "import/resolver": "webpack"
  }
}
```

**`package.json`**

```json
{
  "type": "module",
  "dependencies": {
    "eslint": "8.1.0",
    "eslint-import-resolver-webpack": "0.13.2",
    "eslint-plugin-import": "2.25.2",
    "katex": "0.15.1",
    "webpack": "5.61.0",
    "webpack-cli": "4.9.1"
  }
}
```

```console
$ npm i
…

$ node src.js
Hello KaTeX 0.15.1

$ npx webpack --mode=production
…
webpack 5.61.0 compiled with 3 warnings in 2988 ms

$ node dist/main.js 
Hello KaTeX 0.15.1

$ npx eslint src.js

/tmp/test/src.js
  1:19  error  Unable to resolve path to module 'katex'  import/no-unresolved

✖ 1 problem (1 error, 0 warnings)
```

**Comment by ljharb** [comment]
The issue is that `katex` has entry points defined in `exports` that do not, for backwards compatibility, have corresponding files on the filesystem for older nodes to resolve.

This package uses `resolve`, and neither has support yet for `exports`. So, either `katex` will have to add those files (which it should anyways, since it doesn't declare `engines.node`, which means it probably works on every node version), or, you'll need to use `import/ignore` on that dependency until we're able to add that support.
